### ISSUE TITLE
Typecode not supported in cluster

### ISSUE THREAD

**Issue by stephenlautier** [dotnet/orleans#5993]
We recently updated to Orleans 2.4.2 + update SignalR.Orleans as well, and in certain cases we are getting error such as `TypeCode $1903070868 not supported in the cluster`

![image](https://user-images.githubusercontent.com/3908723/65983209-4f0fef00-e47d-11e9-8fb2-b4489ecaa00a.png)

We should only have 1 cluster and all silos are all the same build and we dont use grain versioning either; Seeing slightly the code thats why it seems to be thrown if im correct, however in our case it shouldn't be the case

**Comment by Frank-Krick** [comment]
I see the same error oin `3.0.0-beta1` during Register as stream producer:
```
RegisterAsStreamProducer failed due to Orleans.Runtime.OrleansException: TypeCode $1903070868 not supported in the cluster
```

Btw. is there a way to get access to the `Catalog` so that I could get more information about the failing grain using `GetGrainTypeInfo`? Or is there another method to get more information about the grain from the type code?

**Comment by stephenlautier** [comment]
Not sure if this is the main reason to this, with a case like this should it work correctly?

- App host (project) had a dependency on Orleans e.g. 2.4.2 (and has grains with codegen)
- App depends on a library with Grains with Orleans 2.3.4 (with codegen)

If it should, it seems that when we have a dependency with a Grain (codegen) which the version of Orleans doesn't match ours, we either:
- Get this issue (like the issue posted)
OR 
- Extension not installed on grain  e.g. 
>Extension not installed on grain SignalR.Orleans.Groups.GroupGrain attempting to invoke type Orleans.Streams.OrleansCodeGenStreamConsumerExtensionMethodInvoker from invokable Orleans.Runtime.ActivationData Such as this https://github.com/OrleansContrib/SignalR.Orleans/issues/89

**Comment by MV10** [comment]
I've started getting intermittent `Typecode $xxxxx not supported in the cluster` on a demo/POC-style project. The project was shelved for about six months, and when I spun it up again I bumped all the Orleans packages up to the latest (as of last month). This is the only discussion I've found of that error message. It sounds as if the thinking here is that something goes wrong after an Orleans update?

How does one work out what the `$xxxxx` actually refers to?

(Unfortunately I can't post detailed logs -- those are locked away on a highly isolated development VM.)
